src/jarabe/journal/listmodel.py

```diff
@@ -92,8 +90,19 @@ def __init__(self, query):
         # avoid hitting D-Bus and disk.
         self.view_is_resizing = False
 
+    def set_query(self, query, clear_selection=True):
```

Comment:
Sorry @samdroid-apps I am slow with reviews this week. Could you prepare a new pr with the patch in #520, and this one modified to make easier test all together?